device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/ModuleClient.java

```diff
@@ -117,7 +117,7 @@ public ModuleClient(String connectionString, IotHubClientProtocol protocol, Clie
      * @param sasTokenProvider The provider of all SAS tokens that are used during authentication.
      * @param protocol The protocol that the client will connect over.
      */
-    public ModuleClient(String hostName, String deviceId, String moduleId, SasTokenProvider sasTokenProvider, IotHubClientProtocol protocol)
+    private ModuleClient(String hostName, String deviceId, String moduleId, SasTokenProvider sasTokenProvider, IotHubClientProtocol protocol)
```

Comment:
Seems like this might be something potentially used by customers currently. Unless we're explicitly not allowing  SAS token regeneration anymore?

What is the benefit of removing this constructor?

Reply:
Good catch, this should remain public. I'm guessing we just weren't using it in an e2e test so it was flagged as a method that could have its scope reduced